### ISSUE TITLE
Implement 'P1855R0 Make `<compare>` freestanding'

### ISSUE THREAD

**Issue by cjdb** [llvm/llvm-project#100030]

**Comment by frederick-vs-ja** [comment]
Subtask of #78350? (It's itself trivial to me to make `<compare>` freestanding.)

**Comment by cjdb** [comment]
> Subtask of #78350? (It's itself trivial, or even "nothing to do", to me to make `<compare>` freestanding.)

Yes, we'll need to make a project for the freestanding effort, in addition to several others.
